File: lib/layer2_core/dark/e_on_primary_fixed_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnPrimaryFixedDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd22;
      case 'standard': return cstd22;
      case 'blue': return cbd22;
      case 'green': return cgd22;
      case 'red': return crd22;
      case 'nothing': return cnod22;
      default: return cstd22;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnPrimaryFixedDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnPrimaryFixedDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd22
            AppTheme.STANDARD -> R.color.cstd22
            AppTheme.BLUE -> R.color.cbd22
            AppTheme.GREEN -> R.color.cgd22
            AppTheme.RED -> R.color.crd22
            AppTheme.NOTHING -> R.color.cnod22
        }
    }
}
